pandas/core/arrays/timedeltas.py

```diff
@@ -217,6 +217,32 @@ def _simple_new(
         result._dtype = TD64NS_DTYPE
         return result
 
+    @classmethod
+    def _from_sequence_strict(
+        cls, data, *, dtype=TD64NS_DTYPE, copy: bool = False
+    ) -> "TimedeltaArray":
+        # GH#37179 eventually we want _from_sequence to be strict
+        if dtype:
+            _validate_td64_dtype(dtype)
+
+        data, copy = dtl.ensure_arraylike(data, copy)
+
+        if data.dtype.kind == "m":
+            pass
+        elif data.dtype == object:
+            inferred = lib.infer_dtype(data)
+            if inferred in ["timedelta64", "timedelta", "empty"]:
+                pass
+            else:
+                raise TypeError(inferred)
```

Comment:
Wondering, should this be a ValueError instead of TypeError? For typical construction we use ValueError (eg `pd.Series(["a"], dtype=float)`), but of course in those cases there is not a single expected type, so ValueError might make more sense there (eg casting from one type to another gives TypeError if it's not a supported cast).

Reply:
sure